auth/store.go

```diff
@@ -879,7 +883,7 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}) *authStore {
+func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}, clientCertAuthEnabled bool) *authStore {
```

Comment:
Yes, on the second thought, the condition based on `Cfg.ClientCertAuthEnabled` seems to be not so meaningful. I'd like to remove it. Maybe making a clear rule about priorities of auth methods (a token generated by `Authenticate()` and common name of cert) and documenting it is required.